Update changelog for v1.1.0 (#2590)
spiffe/spire: CHANGELOG.md

# Changelog

## [1.1.0] - 2021-10-10

### Added
- SPIRE images are now published to GitHub Container Registry. They will continue to be published to Google Container Registry for the next release (#2576,#2580)

> - SPIRE Server now implements the TrustDomain API, which allows callers to manage dynamic federation relationships (#2497,#2524,#2532,#2533,#2535,#2536,#2541,#2544,#2545,#2546,#2547,#2549,#2550,#2557,#2559,#2560,#2561,#2562,#2563,#2564,#2565,#2566,#2568,#2571,#2572,#2574,#2575).

Reviewer: > rephrasing the changelog entry to something that includes the CLI functionality

From user perspective, the CLI support and API support actually feel like two different things and maybe should have their own lines in the changelog. Perhaps that is where some of the tension is coming?

I think the project is what we really want to reference here but I'm afraid about breaking the consistency as @amartinezfayo points out. Maybe we should just leave it.

> - SPIRE Server now implements the TrustDomain API, which allows callers to manage dynamic federation relationships (#2497,#2524,#2532,#2533,#2535,#2536,#2541,#2544,#2545,#2546,#2547,#2549,#2550,#2557,#2559,#2560,#2561,#2562,#2563,#2564,#2565,#2566,#2568,#2571,#2572,#2574,#2575).

Author: Leave it which way? (link to PRs, link to proto, link to project)?

- The TPM DevID Node Attestor emits a new `issuer:cn` selector with the common name of the issuing certificate (#2581)
- The K8s Bundle Notifier plugin now supports pushing the bundle to resources in multiple clusters (#2531)
- The SVIDStore plugin type has been introduced to enable, amongst other things, agentless workload scenarios (#2176,#2483)

> - A built-in AWS Secrets Manager SVIDStore plugin has been introduced, which can push workload SVIDs into AWS secrets (#2542)

Reviewer: The real goal of this work is to enable agentless workloads, and specifically serverless workloads. IMO we should strongly point folks towards an agent where they can afford one. In alignment with that philosophy, the below recommendation suggests that we clarify the purpose of this plugin in an effort to reduce its general use in circumstances where an agent is the better option

```suggestion
- A built-in AWS Secrets Manager SVIDStore plugin has been introduced, which can push workload SVIDs into AWS secrets for use in Lambda functions etc.(#2542)
```

Author: Good suggestion

- The agent and entry list commands in the CLI gained additional filtering capabilities (#2478,#2479)

### Changed

> - With the GA release of GCP CAS, the UpstreamAuthority plugin now needs to know which pool the CA belongs to. If not configured, it will do a pessimistic scan of all pools to locate the correct CA. This scan will be removed in a future release (#2569)

Reviewer: I think this might justify a line in "added" new configurable for ...

I know it's a duplication, but given that "added" is the first section and often the one people are most excited to read, it feels to me like new configurables introduced in favor of old ones or whatever should go there too (if anything to just call attention to them)

Author: 👍

- The K8s Workload Registrar now supports Kubernetes 1.22 (#2515,#2540) 

> - Self-signed CA certificates now have a random serial number (#2494)

Reviewer: nit: I think many users will read this statement and worry about its impact and what it will mean in their environment. Since we can say with relative confidence "it's no problem", my vote would be to tone it down a little such that it carries a "we're now normal" quality

```suggestion
- Self-signed CA certificate serial numbers are now conformant to RFC 5280 (#2494)
```

Author: 👍

- The AWS KMS Key Manager plugin now creates keys with a very strict policy by default (#2424)

> - SPIRE Server CLI commands now page results from the API (#2456)

Reviewer: What does this mean for the user? I can't say this is accurate, but something like:

```suggestion
- SPIRE Server CLI list/show commands now work reliably for deployments greater than XYZ things (#2456)
```

- The deprecated agent key file (`svid.key`) is proactively removed by the agent. It was only maintained to accomodate rollback from v1.0 to v0.12 (#2493)

### Removed
- The deprecated `service_account_whitelist` configurables have been removed from the SAT and PSAT Node Attestor plugins (#2543)
- The deprecated `projectid_whitelist` configurable has been removed from the GCP IIT Node Attestor plugin (#2492)
- The deprecated `bundle_endpoint` and `registration_uds_path` configurables have been removed from SPIRE Server (#2486,#2519)

> - Legacy (v0) plugin support has been removed. All plugins must now be authored using the plugin SDK. (#2488,#2490,#2516,#2517,#2518,#2520,#2521,#2522,#2523,#2525,#2528)

Reviewer: Similar comment here about most significant change(s) referenced

Author: 👍

> - Support for the deprecated Registration API has been removed (#2487)

Reviewer: 🎉
